mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -172,7 +169,7 @@ def DeallocQubitOp : Memory_Op<"dealloc_qb"> {
     }];
 }
 
-def DeallocOp : Memory_Op<"dealloc"> {
+def DeallocOp : Memory_Op<"dealloc", [MemoryEffects<[MemFree<QuantumMemory>]>]> {
```

Comment:
The asymmetry between the alloc and dealloc really bothers me, but I guess it was one way achieve what we wanted (which was being able to DCE allocations if they are unused, but maintain deallocs as a valid use that should never be deleted).